Makefile.dep

```diff
@@ -184,6 +184,10 @@ ifneq (,$(filter trickle,$(USEMODULE)))
   USEMODULE += xtimer
 endif
 
+ifneq (,$(filter eui_provider,$(USEMODULE)))
+  USEMODULE += luid
+endif
```

Comment:
All this `eui_provider` stuff seems out of place in a PR that is about keeping track of `netdev`s. Maybe move to another PR that depends on this PR or is dependent on this PR?

Reply:
Ok, moved that to another PR.
May I squash while I remove those commits from this PR?

Reply:
Yepp